src/wasm-binary.h

```diff
@@ -734,7 +734,8 @@ class WasmBinaryWriter {
   MixedArena allocator;
 
   Function::DebugLocation lastDebugLocation;
-  size_t lastBytecodeOffset;
+  std::vector<std::pair<size_t, const Function::DebugLocation*>> sourceMapLocations;
```

Comment:
please add a comment explaining why we need this, maybe `we store source map locations until the section is placed at its final location (shrinking LEBs may cause changes there)`